client/README.md

```diff
@@ -1,208 +1,389 @@
+    - [Unqualified References](#unqualified-references)
+- [Pre and Postprocessing](#pre-and-postprocessing)
+  - [Handling Macro/Templating Languages](#handling-macrotemplating-languages)
+  - [Custom Pre and Postprocessing Logic](#custom-pre-and-postprocessing-logic)
+- [Renaming SQL Identifiers](#renaming-sql-identifiers)
+- [Deploying to Cloud Run](#deploying-to-cloud-run)
+
+## Quickstart
+
+```shell
+# Install the CLI as described in the Installation section below.
+
+# Copy the example project directory to a project directory of your own.
+cp -R examples/teradata/sql <YOUR_PROJECT_DIRECTORY>
+
+# Change directory to your project directory.
+cd <YOUR_PROJECT_DIRECTORY>
+
+# Remove the example input files from the input directory.
+rm -rf input/*
+
+# Copy the files you would like to translate into the input directory.
+cp <YOUR_INPUT_FILES> input/
+
+# Edit the config/config.yaml file appropriately as described in the Basic Usage
+# section below.
+nano config/config.yaml
+
+# Set the appropriate environment variables as described in the Basic Usage
+# section below.
+export BQMS_VERBOSE="False"
+export BQMS_MULTITHREADED="True"
+export BQMS_PROJECT="<YOUR_GCP_PROJECT>"
+export BQMS_GCS_BUCKET="<YOUR_GCS_BUCKET>"
+
+# Run the translation.
+./run.sh
 ```
-pip3 install dwh-migration-tools/client
-```
-
-Install the gcloud CLI following the [instructions](http://cloud.google.com/sdk/docs/install).
-
-### [Optional] gcloud login and authentication
 
-The program will first validate the login and credential status of
-gcloud. If the validation steps failed, the program will run the following two
-commands automatically and bring up a page on browser for your agreement of
-using your Google account.
+## Installation
 
-However, it's recommended that you run these two commands on the terminal first as a one-time setup requirement:
+Prerequisites: Python >= 3.7.2.
 
-Log in to gcloud:
+Preferred OS: Linux or MacOS.
 
-```
-gcloud auth login
+```shell
+pip install .
 ```
 
-Generate an application-default credential file so that you can use gcloud API
-programmatically:
-
-```
-gcloud auth application-default login
+## Basic Usage
+
+There are two ways to run the CLI: directly and via a wrapping  `run.sh` script.
+
+Direct execution requires setting
+[environment variables](#environment-variables) and modifying a
+[config.yaml](#configyaml) file.
+
+`run.sh` execution is described in detail in 
+[Recommended Usage](#recommended-usage).
+
+### Environment Variables
+
+The CLI accepts the following environment variables:
+
+  - `BQMS_PROJECT` **(required)**: The project that the translation job will be
+    run in. All GCS paths must reside in this project.
+  - `BQMS_INPUT_PATH` **(required)**: The path that raw input to be preprocessed
+    will be read from . This can either be a local path or a path that 
+    begins with the `gs://` scheme.
+  - `BQMS_PREPROCESSED_PATH` **(required)**: The path that the batch SQL
```

Comment:
Do we need to add these required filed in the quickstart section?

Reply:
The quickstart uses `run.sh` not the Python CLI directly. These env vars are only required for the Python CLI. `run.sh` sets these env vars for you automatically based on the example project structure.